### PR TITLE
Hotfix remove geograhicalArea import contraint

### PR DESCRIPTION
**Pull request by nick-beyondessential**
### Issue #: [2482](https://github.com/beyondessential/tupaia-backlog/issues/2482) Issue with importing entities

- remove contraint on parents having geographicalAreas recently added for country level facilities

### CODE REVIEW

File: packages/data-broker/src/services/dhis/DhisInputSchemeResolvingApiProxy.js [hunk 1]
```diff
@@ -97,7 +97,7 @@ export class DhisInputSchemeResolvingApiProxy {
 
     for (const dataElementCode of dataElementCodes) {
       const dataElement = dataElements.find(d => d.code === dataElementCode);
-      if (!dataElement.config.dhisId) {
+      if (!dataElement?.config?.dhisId) {
```
Comment (edmofro) [inline packages/data-broker/src/services/dhis/DhisInputSchemeResolvingApiProxy.js:100]: I think you've got the wrong base branch @nick-beyondessential?

Comment (nick-beyondessential) [reply packages/data-broker/src/services/dhis/DhisInputSchemeResolvingApiProxy.js]: yes, just changed

File: packages/meditrak-server/src/routes/importEntities/getOrCreateParentEntity.js [hunk 2]
```diff
@@ -112,10 +112,8 @@ export async function getOrCreateParentEntity(transactingModels, entityObject, c
       parentEntity,
       transactingModels,
     );
-    if (!parentGeographicalArea)
-      throw new Error(
-        `Parent entity must have geographical area, parent entity code: ${parentCode}`,
-      );
+    if (!parentGeographicalArea) return { parentEntity };
```
Comment (edmofro) [inline packages/meditrak-server/src/routes/importEntities/getOrCreateParentEntity.js:115]: Should still throw the error if entity is of type facility

**Review by edmofro — CHANGES_REQUESTED** [review]

File: packages/meditrak-server/src/routes/importEntities/getOrCreateParentEntity.js (hunk 2 above)
Comment (edmofro) [reply packages/meditrak-server/src/routes/importEntities/getOrCreateParentEntity.js:115]: If you look at the calling code, this is the only place that parent geographical area is used
```
const { parentGeographicalArea, parentEntity } = await getOrCreateParentEntity(
      transactingModels,
      entityObject,
      country,
    );
    if (entityType === transactingModels.entity.types.FACILITY) {
      const defaultTypeDetails = getDefaultTypeDetails(facilityType);
      const facilityToUpsert = {
        type: facilityType,
        type_name: typeName || undefined, // Ensure empty string is treated as undefined
        category_code: categoryCode || undefined, // Ensure empty string is treated as undefined
        code,
        name,
        geographical_area_id: parentGeographicalArea.id,
      };
```

The validation could also be done there instead.

**Review by edmofro — APPROVED** [review]
Thanks (and glad that linting got caught)

File: packages/meditrak-server/src/routes/importEntities/updateCountryEntities.js [hunk 3]
```diff
@@ -31,19 +31,24 @@ const getEntityMetadata = async (transactingModels, code, pushToDhis) => {
   const defaultMetadata = {
     dhis: { isDataRegional: true },
   };
-  
+
```
Comment (nick-beyondessential) [inline packages/meditrak-server/src/routes/importEntities/updateCountryEntities.js:34]: caught a bunch of eslint auto-tidyups

**Comment by nick-beyondessential** [comment]
Thanks for the review despite your busy schedule @edmofro, appreciate it!

**Pull request merged**